src/core/atomic.d

```diff
@@ -190,6 +222,14 @@ else version( AsmX86_32 )
         //
         // +=   -=  *=  /=  %=  ^^= &=
         // |=   ^=  <<= >>= >>>=    ~=
+        static if( op == "+=" && __traits(isIntegral, T) && T.sizeof <= 4) {
+            return atomicFetchAdd!(T)(val, mod);
+        }
+        else
+        static if( op == "-=" && __traits(isIntegral, T) && T.sizeof <= 4) {
+            return atomicFetchAdd!(T)(val, -mod);
```

Comment:
How about adding a private `atomicSubFetch` that simply calls `atomicFetchAdd!(T)(val, -mod)`?
That way the primitive is encapsulated and the GDC/LDC people can easily map it to their intrinsics.